pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -534,6 +542,8 @@ const (
 )
 
 // MonSpec represents the specification of the monitor
+// +kubebuilder:validation:XValidation:message="zones must be less than or equal to count",rule="!has(self.zones) || (has(self.zones) && (size(self.zones) <= self.count))"
```

Comment:
For stretch cluster we can verify specific values, which will simplify these rules as well.
- mon.count must be 5
- The number of zones must be exactly 3
- Exactly one zone should have `arbiter.true`

Reply:
This will be new rule, right?

Reply:
These two rules about the mon count for stretch cluster don't make sense to me. I would suggest removing those two rules and adding the rules I mentioned above.

Reply:
the rules I have taken is from the existing webhook rules https://github.com/rook/rook/blob/510d2721b4ca350f26c998d528926eb1e3b66e4f/pkg/apis/ceph.rook.io/v1/cluster.go#L49-L61 PTAL and see if this rule and webhook matches and make sense.

What you suggesting is also doable but seems like different than what we already have in webhook checks

Reply:
Oh I see now, this rule was intended for adding zones for non-stretch clusters in #12384. Let's keep this first rule for that case. But the stretch clusters always require exactly 3 zones. So the second rule (line 546) could check for exactly 3 zones.

Reply:
added